instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java

```diff
@@ -98,4 +104,25 @@ public static void buildAndFinishSpan(Context context, PubsubMessage pubsubMessa
     Context current = subscriberInstrumenter.start(newContext, pubsubMessage);
     subscriberInstrumenter.end(current, pubsubMessage, null, null);
   }
+
+  public static Optional<Object> extractPubsubMessageAttributes(PubsubMessage pubsubMessage) {
+    try {
+      Class cls = pubsubMessage.getClass();
```

Comment:
Why not create a "getAttributeWithReflection" method? Receives an object, attribute name?

Reply:
it's not that generic.. it's very specific for pubsubmessage

Reply:
What about something like this?

`public static extractAttributeFromObject(Object object, String attrName) {
	Class cls = object.getClass();
	Field attribute = cls.getDeclaredField(attrName);
	attribute.setAccessible(true);
	return attribute.get(object);
}`